diff --git a/runtime/src/support/Any.h b/runtime/src/support/Any.h
index 5db59f6..60699f9 100644
--- a/runtime/src/support/Any.h
+++ b/runtime/src/support/Any.h
@@ -117,21 +117,9 @@ private:
 
     T value;
 
-    Base* clone() const {
-      return clone<>();
-    }
-
-  private:
-    template<int N = 0, typename std::enable_if<N == N && std::is_nothrow_copy_constructible<T>::value, int>::type = 0>
     Base* clone() const {
       return new Derived<T>(value);
     }
-
-    template<int N = 0, typename std::enable_if<N == N && !std::is_nothrow_copy_constructible<T>::value, int>::type = 0>
-    Base* clone() const {
-      return nullptr;
-    }
-
   };
 
   Base* clone() const
